api/python/provisioner/commands/deploy_vm.py

```diff
@@ -252,16 +252,25 @@ def run(self, **kwargs):
                         f"Mounting partition {metadata_device} "
                         f"into {mount_point} (with fstab record)"
                     )
-                    function_run(
-                        'mount.mounted',
-                        fun_kwargs=dict(
+                    fun_kwargs = dict(
                                    name=mount_point,
                                    device=metadata_device,
                                    fstype='ext4',
                                    mkmnt=True,
-                                   persist=True),
-                        targets=self._primary_id()
-                    )
+                                   persist=True)
+                    if self.setup_ctx:
+                        self.setup_ctx.ssh_client.run(
+                            'state.single',
+                            fun_args=['mount.mounted'],
+                            fun_kwargs=fun_kwargs,
+                            targets=self._primary_id()
+                        )
+                    else:
+                        function_run(
```

Comment:
there is no such function in `mount` salt's module, `mount.mounted` is a state. Please double check the example I posted before.

Reply:
comment is addressed